webcompat/webhooks/helpers.py

```diff
@@ -27,6 +27,7 @@ def extract_browser_label(body):
         # right now, match_list.groups(0) should look like:
         # ('browser', 'firefox')
         browser = match_list.groups(0)[1].lower()
+	browser = browser.translate(None, "()")
```

Comment:
A couple of changes here mostly syntactic.

1. The indentation is not correct. Python is strict about indentation for defining blocks of code. As it is it will break. Make sure to indent with spaces.
2. Use simple quotes around the paranthesis. `translate(None, '()')`
3. Big +1 on using `translate` instead of `replace` (this is faster).

Reply:
Hey @karlcow Thanks a lot :) And yeah I actually don't know what exactly this test, you're talking about is for ? Its i guess due to the reason that I don't know what these functions are doing ! A help is needed for sure.

Reply:
so usually it is better to start with tests to know what the function should do with the data. Let's say you are at the root of the project:

```bash
# Let's now start python IDLE
python
# Python 2.7.10 (default, Feb  7 2017, 00:08:15) 
# [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
# Type "help", "copyright", "credits" or "license" for more information.
# >>> 
```

Now we are in the python environment, to quit you can just type ctrl+D anytime, but for now let's continue. In `webhooks/helpers.py`, we want to use the extract browser label function. This is the current form before your modifications.

https://github.com/webcompat/webcompat.com/blob/18c327aedf76aae5207b0bf69fcb33bd6ac90d1f/webcompat/webhooks/helpers.py#L16-L33

```python
>>> from webcompat.webhooks.helpers import extract_browser_label
>>> # as you can see the function is taking a body argument. Let's make one.
>>> body = "<!-- @browser: Firefox Mobile 40.0 -->"
>>> # Now we can pass this body to the function we imported:
>>> extract_browser_label(body)
'browser-firefox-mobile'
>>> # We can see how it extracted the @browser information.
... # And made this as a string with dashes.
>>> # What if the body contains something with parenthesis.
>>> body = "<!-- @browser: Firefox Mobile (Tablet) 40.0 -->"
>>> # Let's try again!
... extract_browser_label(body)
'browser-firefox-mobile-(tablet)'
>>> # OK so this is the issue we are trying to solve.
>>> # In this case we want the extract_browser_label(body) function
>>> # to return: 'browser-firefox-mobile-tablet'
>>> # This is our test.
```

Let's go back to https://github.com/webcompat/webcompat.com/blob/ffb0dc3932871c7576cbfb9bc82f76c024464983/tests/test_webhook.py#L119-L124

```python
    def test_extract_browser_label(self):
        """Extract browser label name."""
        browser_label = helpers.extract_browser_label(self.issue_body)
        self.assertEqual(browser_label, 'browser-firefox')
        browser_label_none = helpers.extract_browser_label(self.issue_body2)
        self.assertEqual(browser_label_none, None)
        browser_label_paren = helpers.extract_browser_label(
            '<!-- @browser: Firefox Mobile (Tablet) 40.0 -->')
        self.assertEqual(browser_label_paren, 'browser-firefox-mobile-tablet')
```

by having this third test we make sure that the project will not fail in the future if we modify something in the function. If I run this test with the current structure before your modification, we get:

```bash
nosetests -v --nocapture tests/test_webhook.py
# Extract browser label name. ... FAIL
# POST without bogus signature on labeler webhook is forbidden. ... ok
# POST with event not being 'issues' or 'ping' fails. ... ok
# POST without signature on labeler webhook is forbidden. ... ok
# POST with an unknown action fails. ... ok
# GET is forbidden on labeler webhook. ... ok
# Extract the right information from an issue. ... ok
# Validation tests for GitHub Webhooks. ... ok
# POST with PING events just return a 200 and contains pong. ... ok
# 
# ======================================================================
# FAIL: Extract browser label name.
# ----------------------------------------------------------------------
# Traceback (most recent call last):
#   File "/Users/karl/code/webcompat.com/tests/test_webhook.py", line 127, in test_extract_browser_label
#     self.assertEqual(browser_label_paren, 'browser-firefox-mobile-tablet')
# AssertionError: 'browser-firefox-mobile-(tablet)' != 'browser-firefox-mobile-tablet'
# 
# ----------------------------------------------------------------------
# Ran 9 tests in 0.056s
# 
# FAILED (failures=1)
```

The test is failing because indeed we don't take care of the parenthesis. Let's use the function you have done.


```python
def extract_browser_label(body):
    """Parse the labels from the body in comment:

    <!-- @browser: value -->.
    Currently this only handles a single label,
    because that's all that we set in webcompat.com.
    """
    match_list = re.search(r'<!--\s@(\w+):\s([^\d]+?)\s[\d\.]+\s-->', body)
    if match_list:
        # perhaps we do something more interesting depending on
        # what groups(n)[0] is in the future.
        # right now, match_list.groups(0) should look like:
        # ('browser', 'firefox')
        browser = match_list.groups(0)[1].lower()
	browser = browser.translate(None, "()")
        dash_browser = '-'.join(browser.split())
        return 'browser-{name}'.format(name=dash_browser)
    else:
        return None
```

Let's run the test again

```bash
nosetests -v --nocapture tests/test_webhook.py
# Extract browser label name. ... ok
# POST without bogus signature on labeler webhook is forbidden. ... ok
# POST with event not being 'issues' or 'ping' fails. ... ok
# POST without signature on labeler webhook is forbidden. ... ok
# POST with an unknown action fails. ... ok
# GET is forbidden on labeler webhook. ... ok
# Extract the right information from an issue. ... ok
# Validation tests for GitHub Webhooks. ... ok
# POST with PING events just return a 200 and contains pong. ... ok
# 
# ----------------------------------------------------------------------
# Ran 9 tests in 0.054s
# 
# OK
```

Working. No test failure.

I hope it helps.
Could you modify your code so to add the requested changes. Thanks.